Geta.EPi.Extensions/UrlExtensions.cs

```diff
@@ -75,5 +75,63 @@ private static Uri GetBaseUri()
                 ? SiteDefinition.Current.SiteUrl
                 : new Uri(HttpContext.Current.Request.Url.GetLeftPart(UriPartial.Authority));
         }
+
+        /// <summary>
+        ///     Gets the Url type
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns>The type of Url (Internal, External, Media). Returns Unknown if resolution fails.</returns>
+        public static UrlType GetUrlType(this Url url)
+        {
+            if (url.IsAbsoluteUri)
+            {
+                if (SiteDefinition.Current == null)
+                {
+                    // if we fail to get site definition (for instance in scheduled jobs) we assume that link is external
+                    return UrlType.External;
+                }
+
+                return SiteDefinition.Current.Hosts.Any(h => h.Name.Equals(url.Host, StringComparison.InvariantCultureIgnoreCase)) ? UrlType.Internal : UrlType.External;
```

Comment:
Now it has another problem - it is not consistent. For example, "http://myurl.com/media.jpg" and "/media.jpg" won't return the same result. First will return "UrlType.Internal", second will return "UrlType.Media" despite it is the same content.